src/utils/chartOptionsHelpers.js

```diff
@@ -81,7 +80,7 @@ export const plotNullDataLayerToAxis = (xAxis, series, broadcastSetState) => {
         events: {
           mouseover: function(e) {
             broadcastSetState &&
-              broadcastSetState({'customLegend': createCartesianCustomLegendData(this.axis.series, null)});
+              broadcastSetState({'customLegend': createCartesianCustomLegendData(this.axis.series, idx)});
```

Comment:
this will work. but i wonder if `this.index` is also available here? could be more semantic?